pkg/apis/kubernetes/v1/common.go

```diff
@@ -0,0 +1,56 @@
+package v1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/apis/internal"
+)
+
+const awaitCompletionPollInterval = 30 * time.Second
+
+var (
+	// ErrStateError is returned if a resource could not be provisioned (state "Error")
+	ErrStateError = errors.New("resource has state \"Error\"")
+
+	// ErrStateUnknown is returned if a resource has an unknown state
+	ErrStateUnknown = errors.New("resource has an unknown state")
+)
+
+type objectWithState interface {
+	types.Object
+	getState() internal.State
+}
+
```

Comment:
we already have this with `lbaasv1.StateRetriever` and I'd like to reuse that interface (and move it out of `lbaasv1`)